subsys/bluetooth/mesh/net.c

```diff
@@ -772,7 +772,8 @@ int bt_mesh_net_resend(struct bt_mesh_subnet *sub, struct net_buf *buf,
 	bt_mesh_adv_send(buf, cb, cb_data);
 
 	if (!atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS) &&
-	    bt_mesh.seq > IV_UPDATE_SEQ_LIMIT) {
+	    bt_mesh.seq > IV_UPDATE_SEQ_LIMIT &&
+	    bt_mesh_subnet_get(BT_MESH_KEY_PRIMARY)) {
```

Comment:
@trond-snekvik you're right, that's an oversight - the most natural place is the central place where we increment the sequence number, i.e. `bt_mesh_seq_next()` I'll submit an improved patch soon. I think the reason this hadn't been noticed is that it has never been properly tested - for qualification purposes the test APIS are used which force an IV update on demand.

Reply:
Yeah, we've struggled with this part as well, the test spec is a Swiss cheese at best... Things like this will probably only ever be testable with a proprietary unit test. 

If you want to make a separate PR for the fix, this one looks ready to go in.

Reply:
@trond-snekvik I decided to just do it in one go, please take a look if it makes sense. I had to think a bit that this doesn't break anything since now the IV Update is started slightly earlier in `bt_mesh_net_resend()`, however e.g. the results of encoding/encrypting messages are the same since `BT_MESH_NET_IVI_TX` keeps the same value in the beginning of the IVU process. For a moment I thought about deferring through k_work, but it should be safe to keep doing this synchronously like I've done. Please take a look if it seems ok.

Reply:
I think you have to either postpone it to a work item, or put the code in front of the seqnum allocation in that function, looks like you'll end up putting an old (high) seqnum with the new iv index in when sending with `bt_mesh_net_send`, as the ivi bit is set after.

Reply:
@trond-snekvik `BT_MESH_NET_IVI_TX` is defined as:
```
#define BT_MESH_NET_IVI_TX (bt_mesh.iv_index - \                                
                            atomic_test_bit(bt_mesh.flags, \                    
                                            BT_MESH_IVU_IN_PROGRESS))           
```

I.e. it'll return the same value both before and starting IV Update. Or do you see some other place where we'd incorrectly pick the new IV Index?

Reply:
@trond-snekvik or were you referring to `bt_mesh_store_seq()`? That uses `k_delayed_work` (see `schedule_store()` in `settings.c`), so the storing doesn't happen synchronously.

Reply:
Damn, you're right, sorry. It's the beacons that change IV index when we enter IVU 🤦‍♂ 

Store seq looks like it should be unaffected by this change.